File: src/classical_ciphers/caesar_cipher.cpp
#include "../../include/classical_ciphers/caesar_cipher.h"
#include <string>

std::string Caesar::encrypt(const std::string &plaintext, const std::string &key) {
    // TODO: input validation
    
    std::string ciphertext;
    int keyInt = std::stoi(key);
    for (char c : plaintext) {
        ciphertext.push_back('a' + (c - 'a' + keyInt) % 26);
    }
    return ciphertext;
}

std::string Caesar::decrypt(const std::string &ciphertext, const std::string &key) {
    // TODO: input validation
    
    std::string plaintext;
    int keyInt = -std::stoi(key);
    while (keyInt < 0) {
        keyInt += 26;
    }
    for (char c : ciphertext) {
        plaintext.push_back('a' + (c - 'a' + keyInt) % 26);
    }
    return plaintext;
}

File: python/src/classical_ciphers/caesar_cipher.py
def encrypt(plaintext, key):
    # Ensure the key is an integer
    key = int(key)
    ciphertext = ""
    for char in plaintext:
        if char.isalpha():
            shifted = ord(char) + key
            if char.islower():
                if shifted > ord('z'):
                    shifted -= 26
                ciphertext += chr(shifted)
            elif char.isupper():
                if shifted > ord('Z'):
                    shifted -= 26
                ciphertext += chr(shifted)
        else:
            ciphertext += char
    return ciphertext

def decrypt(ciphertext, key):
    # Ensure the key is an integer
    key = int(key)
    plaintext = ""
    for char in ciphertext:
        if char.isalpha():
            shifted = ord(char) - key
            if char.islower():
                if shifted < ord('a'):
                    shifted += 26
                plaintext += chr(shifted)
            elif char.isupper():
                if shifted < ord('A'):
                    shifted += 26
                plaintext += chr(shifted)
        else:
            plaintext += char
    return plaintext
